qa/tasks/ragweed.py

```diff
@@ -29,46 +56,21 @@ def download(ctx, config):
     assert isinstance(config, dict)
     log.info('Downloading ragweed...')
     testdir = teuthology.get_testdir(ctx)
-    s3_branches = [ 'master', 'nautilus', 'mimic', 'luminous', 'kraken', 'jewel' ]
     for (client, cconf) in config.items():
-        default_branch = ''
-        branch = cconf.get('force-branch', None)
-        if not branch:
-            default_branch = cconf.get('default-branch', None)
-            ceph_branch = ctx.config.get('branch')
-            suite_branch = ctx.config.get('suite_branch', ceph_branch)
-            ragweed_repo = ctx.config.get('ragweed_repo', teuth_config.ceph_git_base_url + 'ragweed.git')
-            if suite_branch in s3_branches:
-                branch = cconf.get('branch', 'ceph-' + suite_branch)
-            else:
-                branch = cconf.get('branch', suite_branch)
-        if not branch:
-            raise ValueError(
-                "Could not determine what branch to use for ragweed!")
-        else:
+        ragweed_repo = ctx.config.get('ragweed_repo',
+                                      teuth_config.ceph_git_base_url + 'ragweed.git')
+        for branch in get_ragweed_branches(ctx.config, cconf):
             log.info("Using branch '%s' for ragweed", branch)
-        sha1 = cconf.get('sha1')
-        try:
-            ctx.cluster.only(client).run(
-                args=[
-                    'git', 'clone',
-                    '-b', branch,
-                    ragweed_repo,
-                    '{tdir}/ragweed'.format(tdir=testdir),
-                    ],
-                )
-        except Exception as e:
-            if not default_branch:
-                raise e
-            ctx.cluster.only(client).run(
-                args=[
-                    'git', 'clone',
-                    '-b', default_branch,
-                    ragweed_repo,
-                    '{tdir}/ragweed'.format(tdir=testdir),
-                    ],
-                )
+            try:
+                ctx.cluster.only(client).sh(
+                    f'git clone -b {branch} {ragweed_repo} {testdir}/ragweed')
```

Comment:
i this a python3 thing? it failed with a syntax error when trying to test https://github.com/ceph/ragweed/pull/9 for nautilus:
```
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_py2/teuthology/run_tasks.py", line 87, in run_tasks
    manager = run_one_task(taskname, ctx=ctx, config=config)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_py2/teuthology/run_tasks.py", line 65, in run_one_task
    task = get_task(taskname)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_py2/teuthology/run_tasks.py", line 29, in get_task
    module = _import('tasks', module_name, task_name, fail_on_import_error=True)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_py2/teuthology/run_tasks.py", line 53, in _import
    0,
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_py2/virtualenv/local/lib/python2.7/site-packages/gevent/builtins.py", line 96, in __import__
    result = _import(*args, **kwargs)
  File "/home/teuthworker/src/github.com_cbodley_ceph_wip-qa-nautilus-ragweed-pip/qa/tasks/ragweed.py", line 69
    f'git clone -b {branch} {ragweed_repo} {testdir}/ragweed')
                                                            ^
SyntaxError: invalid syntax
```

Reply:
@cbodley it is. please do not pass `-t py2` or use python2 for testing nautilus branch as the qa suites in nautilus is now python3 compatible.

Reply:
how do i avoid py2? i'm not passing that to teuthology-suite:
> $ virtualenv/bin/teuthology-suite -s rgw -m smithi --ceph-repo https://github.com/ceph/ceph.git -c nautilus -n 10 --suite-repo https://github.com/cbodley/ceph.git --suite-branch wip-qa-nautilus-ragweed-pip -p 49 --filter ragweed --limit 1

Reply:
@cbodley your wip-qa-nautilus-ragweed-pip branch is based on top of 6bd64f37f60816660a82db0da2abcc49f30d9ddd. and it does not contain a8bbd7b5460804b5f7f982724e7603f628c51656 which specifies the master branch of teuthology. and i assume you are using sepia lab for launching the test, where, on `teuthology` machine, we have `teuthology_branch: py2` in `/etc/teuthology.yaml`. so you can either rebase your branch onto the latest nautilus or create a copy of `/etc/teuthology.yaml` in your `$HOME` and rename it to `.teuthology.yaml` and then remove `teuthology_branch: py2` from it.